docs/developer/designs/time-aware-fairness/time-aware-fairness.md

```diff
@@ -0,0 +1,219 @@
+Regardless of which backend TSDB is chosen for the storage of resource usage data, it's to be expected that busy, large-scale clusters will generate a lot of it, and there could be limitations to the granularity of data preserved, and thus the accuracy of results achieved. It's assumed that some loss of granularity might be required in order to save data for long periods of time: admins might need to balance the historical time window with the data granularity and the storage & compute resources required from the backend data storage; KAI-scheduler needs to allow the admins to configure those to their liking.
+
+#### Intuitive understanding of fairness over time
+Several parameters and configurations can effect the outcome of the scheduling algorithm: good documentation, and potentially tools, will need to be provided in order to help the admin configure the cluster properly. An intuitive configuration should result in behavior that is predictable in simple scenarios: for example, "Using X GPUs for T hours will result in a project with W over-quota weight getting penalized by a factor of 50%". Several variables can go into achieving this result - we need to map these in a clear way.
+
+### Risks and Mitigations
+
+<!--
+ToDo
+-->
+
+## Design Details
+
+> **⚠️ Work in Progress**  
+> This design is currently under development. The mathematical formulation and implementation details are an initial proposal for further discussion. Feedback and suggestions are welcome.
+
+Accounting service will provide the scheduler with data for each cycle about the usage of resources per queue per resource. A vector will represent the usage of each queue per resource type.  
+The resource division formula for over-quota resource can be found in proportion plugin, resource_division.go:
+```Go
+fairShare := amountToGiveInCurrentRound * (overQuotaWeight / totalWeights)
+```
+This calculation is done for each resource type. The result of this calculation is later capped with the requested resource of the queue, so each round of this calculation for all queues could result in remaining capcacity to divide - this is run in rounds, each time with the updated remaining unsatisfied queues, untill all resources are divided or all queues get their request.
+
+For convenience, we'll write it out as a formula:
+
+$$\text{F} = C \cdot \frac{W}{\sum{W}}$$
+
+Where:
+- **F** is the fair share allocated to a specific queue in the current round
+- **C** is the remaining capacity (max amount to give in current round)
+- **w** is the weight for a specific queue
+- **∑w** is the sum of weights across all competing queues
+
+Let's also mark the normalized weight as:
+
+$$W'_i = \frac{W_i}{\sum{W}}$$
+
+
+### Vacant-adjusted normalized usage
+
+First, we'll define a normalized "usage" score for each queue and resource:
+
+$$U'_i = \frac{U_i}{\sum{U}}$$
+
+We can furthermore consider the **unallocated** resources in the cluster. If we go back to the definition of $U'_i$, we can add:
```

Comment:
Do you also plan to keep historical data for the available resources in the cluster to keep track of what was available at specific time in scalable cluster?